src/sas/sasgui/perspectives/fitting/report_dialog.py

```diff
@@ -73,14 +73,17 @@ def onSave(self, event=None):
 
         # save figures
         pictures = []
```

Comment:
Can use enumerate(sequence) and [list comprehension] to remove a bit of boilerplate:
```
    # save the images for use with pdf writer
    pictures = ['_'.join((basename, url.split(':')[1]) for url in self.fig_urls]
    for i, pic in in enumerate(pictures):
       self.report_list[2][i].savefig(pic)
```
Maybe the following is a bit easier to read:
```
    # save the images for use with pdf writer
    def pic_name(basename, url):
        target = url.split(':')[1]
        return '_'.join((basename, target))
    pictures = [pic_name(basename, url) for url in self.fig_urls]
    for i, pic in in enumerate(pictures):
       self.report_list[2][i].savefig(pic)
```